test/lib/Benchmark.js

```diff
@@ -0,0 +1,259 @@
+/* globals XPCOMUtils, Services, Task */
+"use strict";
+
+const {Cu} = require("chrome");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "Task",
+                                  "resource://gre/modules/Task.jsm");
+
+// The default benchmark time is 1 second, i.e. 1e6 microseconds
+const DEFAULT_BENCH_TIME = 1e6;
+
+// For the performance.now(), so the timer resolution is 1 microsecond
+const performance = Services.appShell.hiddenDOMWindow.performance;
+
+// The max iteration number is 1e6 as the timer resolution is 1 microsecond
+const MAX_ITERATION = 1e6;
+
+/*
+ * Get the DOMHighResTimeStamp in microsecond
+ * See more detail at https://developer.mozilla.org/en-US/docs/Web/API/Performance/now
+ */
+function _performanceNow() {
+  return performance.now() * 1e3;
+}
+
+/*
+ * A simple benchmark framework for Activity Stream that runs on top of jpm test
+ *
+ * If you're new to Javascript benchmarking, here is a list of references:
+ *
+ *   https://developer.mozilla.org/en-US/docs/Mozilla/Benchmarking
+ *   https://mathiasbynens.be/notes/javascript-benchmarking
+ *   http://ejohn.org/blog/javascript-benchmark-quality/
+ *
+ * Suitable use cases:
+ *   * measure the load latency of Activity Stream with different profiles and browsering history
```

Comment:
One thing the comment could suggest would be adding performance telemetry for everything that we benchmark so that we can also see distributions and averages in the wild.